cpp/tests/replace/replace_tests.cu

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2018-2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/replace.hpp>
+
+#include <cudf/cudf.h>
+
+// This is the main test fixture
+template <class T>
+struct ReplaceTest : public cudf::test::BaseFixture
+{
+
+  ReplaceTest()
+  {    
+  }
+
+  ~ReplaceTest()
+  {
+  }
+};
+
+template <typename T>
+void normalize_nans_and_zeros_test_internal( cudf::test::fixed_width_column_wrapper<T> const& test_data_in,
+                                             cudf::column_view const& test_data_comp)
+{
+   // mutable overload      
+   {                             
+      cudf::column test_data(test_data_in);
+      cudf::mutable_column_view mutable_view = test_data;
+      cudf::normalize_nans_and_zeros(mutable_view);      
+      
+      // compare bitwise
+      cudf::test::expect_equal_buffers(mutable_view.head(), test_data_comp.head(), mutable_view.size() * sizeof(T));
+   }
+      
+   // returned column overload
+   {
+      cudf::column test_data(test_data_in);
+      cudf::column_view view = test_data;
+      
+      auto out = cudf::normalize_nans_and_zeros(view);      
+      auto out_view = out->view();
+      
+      // compare bitwise
+      cudf::test::expect_equal_buffers(out_view.head(), test_data_comp.head(), out_view.size() * sizeof(T));            
+   }      
+}
+
+// Test for normalize_nans_and_nulls
+TEST(ReplaceTest, NormalizeNansAndZeros)
```

Comment:
This will actually use your test fixture (which looks to be just a shell anyways). 
```suggestion
TEST_F(ReplaceTest, NormalizeNansAndZeros)
```